main/src/cgeo/geocaching/Trackable.java

```diff
@@ -48,6 +48,36 @@
     private TrackableConnector trackableConnector = null;
 
     /**
+     * Merge data from another Trackable.
+     * Squeeze existing data from the new one.
+     *
+     * @param newTrackable from which to pull informations
+     */
+    public void mergeTrackable(final Trackable newTrackable) {
+        guid = newTrackable.guid.equals("") ? guid : newTrackable.guid;
+        geocode = newTrackable.geocode.equals("") ? geocode : newTrackable.geocode;
+        iconUrl = newTrackable.iconUrl.equals("") ? iconUrl : newTrackable.iconUrl;
+        name = newTrackable.name.equals("") ? name : newTrackable.name;
+
+        type = newTrackable.type == null ? type : newTrackable.type;
```

Comment:
Similarly for all the objects following we can use ObjectUtils.defaultIfNull(...,...).